packages/x-date-pickers-pro/src/PickersRangeCalendarHeader/PickersRangeCalendarHeader.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { styled, useThemeProps } from '@mui/material/styles';
+import { PickersCalendarHeader } from '@mui/x-date-pickers/PickersCalendarHeader';
+import { PickerValidDate } from '@mui/x-date-pickers/models';
+import {
+  PickersArrowSwitcher,
+  useLocaleText,
+  useNextMonthDisabled,
+  usePreviousMonthDisabled,
+  useUtils,
+} from '@mui/x-date-pickers/internals';
+import { PickersRangeCalendarHeaderProps } from './PickersRangeCalendarHeader.types';
+
+type PickersRangeCalendarHeaderComponent = (<TDate extends PickerValidDate>(
+  props: PickersRangeCalendarHeaderProps<TDate> & React.RefAttributes<HTMLDivElement>,
+) => React.JSX.Element) & { propTypes?: any };
+
+const PickersRangeCalendarHeaderContentSingleCalendar = styled(PickersCalendarHeader, {
+  name: 'PickersRangeCalendarHeader',
+  slot: 'ContentSingleCalendar',
+  overridesResolver: (_, styles) => styles.contentSingleCalendar,
+})({}) as typeof PickersCalendarHeader;
```

Comment:
Currently, this feels pretty pointless.
We are not adding any custom styles and there is no option to override these styles via `themeOverrides`. 🙈 🤷

Reply:
This is purely to be consistent with https://github.com/mui/mui-x/pull/11988#discussion_r1482866582
For me, either we register this component to the theme completely (default props and styles overrides), or we do neither of them

If on the other discussion we decide to not register the default props, then I do agree that we should remove this one as well

Reply:
I would say that one of the following approaches makes the most sense
1. Add missing `classes` and `styleOverrides` to complete the default `styled` approach
2. Ditch `styled` `PickersRangeCalendarHeaderContentSingleCalendar` and remove the `overridesResolver` on `MultipleCalendar` as it serves no purpose.

Reply:
I'm fine going with 2 and add the overrides back later if someone feels a pain